queries/talon.scm

```diff
@@ -96,7 +96,13 @@
 ;;!  ^^^^^^^^^^^^^
 (match) @condition
 
-(matches) @condition.iteration
+(
+  (source_file
+    (matches) @condition.iteration @condition @condition.trailing
```

Comment:
I would probably skip the iteration scope and make the domain the entire file. Its just one condition per file?

Reply:
`(matches) @condition.iteration @condition @condition.trailing` looks like the entire matches node is the condition. Shouldn't it be `match` instead then?

Reply:
That is also `condition`.  It is now hierarchical.  It is pretty much the exact same pattern we decided to use for `"name"` in Typescript's `const aaa = 0, bbb = 1;`, where

- each nested assignment (`aaa = 0` and `bbb = 1`) is its own domain
- there is a parent scope whose domain is the entire statement that has multiple content ranges, one per sub assignment
- the iteration scope is the entire statement

This situation is analogous, replacing "nested assignment" with "individual condition" and "statement" with "file"

Reply:
Not sure if this is prefared, but its fine with me.

Reply:
I do wonder if we should simplify things and not support the child conditions here 🤔.  Ie there would just be a single scope `"condition"` that refers to the whole condition block from anywhere in the file, and there is no `"every condition"`

Reply:
Before we merge, any thoughts on my last two comments here?

Reply:
I think a single condition is probably enough. You can use line for a single match.